visitors/http_rest/spec_visitor_default.go

```diff
@@ -0,0 +1,315 @@
+		}
+	case *pb.Primitive_BoolValue:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "BoolValue", node.BoolValue)
+		}
+	case *pb.Primitive_BytesValue:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "BytesValue", node.BytesValue)
+		}
+	case *pb.Primitive_StringValue:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "StringValue", node.StringValue)
+		}
+	case *pb.Primitive_Int32Value:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Int32Value", node.Int32Value)
+		}
+	case *pb.Primitive_Int64Value:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Int64Value", node.Int64Value)
+		}
+	case *pb.Primitive_Uint32Value:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Uint32Value", node.Uint32Value)
+		}
+	case *pb.Primitive_Uint64Value:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Uint64Value", node.Uint64Value)
+		}
+	case *pb.Primitive_DoubleValue:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "DoubleValue", node.DoubleValue)
+		}
+	case *pb.Primitive_FloatValue:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "FloatValue", node.FloatValue)
+		}
+	case *pb.Struct:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m,
+				"Fields", node.Fields, // already handled map[string]*Data
+				"MapType", node.MapType,
+			)
+		}
+	case *pb.MapData:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m,
+				"Key", node.Key,
+				"Value", node.Value,
+			)
+		}
+	case *pb.List:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Elems", node.Elems)
+		}
+	case *pb.Optional:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Value", node.Value)
+		}
+	case *pb.Optional_None:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "None", node.None) // TODO: don't recurse?
+		}
+	case *pb.Optional_Data:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Data", node.Data)
+		}
+	case *pb.OneOf:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Options", node.Options)
+		}
+	// These all have No non-basic types as children, so no further recursion is needed
+	case *pb.MethodID: // FIXME: should we even enter this one? Not used by SpecVisitor
+	case *pb.HTTPMethodMeta:
+	case *pb.HTTPPath:
+	case *pb.HTTPQuery:
+	case *pb.HTTPHeader:
+	case *pb.HTTPCookie:
+	case *pb.HTTPBody:
+	case *pb.HTTPEmpty:
+	case *pb.HTTPAuth:
+	case *pb.HTTPMultipart:
+	case *pb.FormatOption_StringFormat: // only contains a string
+	case *pb.None:
+	default:
+		// Fallback to reflection-based implementation
+		return go_ast.DefaultVisitChildren(ctx, vm, m)
+	}
+	return keepGoing
+}
+
+// Visit the struct members given by <field1> <value1> <field2> <value2>
+func visitStructMembers(ctx Context, vm VisitorManager, inStruct interface{}, fields ...interface{}) Cont {
+	keepGoing := Continue
```

Comment:
I initially wrote a version that just took one field and one value.  And then I decided that was silly, so I made this variadic.  But now I wonder whether Go is actually going to allocate a slice on the heap on every call?  Hopefully not.